communication/state_telemetry.cpp

```diff
@@ -203,6 +213,44 @@ static mav_result_t state_telemetry_set_mode_from_cmd(State* state, mavlink_comm
     return result;
 }
 
+static mav_result_t state_telemetry_set_arm_from_cmd(State* state, mavlink_command_long_t* packet)
+{
+    mav_result_t result = MAV_RESULT_ACCEPTED;
+
+    mav_mode_t new_mode;
+    float arm_cmd = packet->param1;
+
+    if (state->mav_state == MAV_STATE_CALIBRATING)
+    {
+        result = MAV_RESULT_TEMPORARILY_REJECTED;
+        print_util_dbg_print("Currently calibrating, impossible to change mode.\r\n");
+    }
+    else
+    {
```

Comment:
Yes exactly, 
we should not directly write 
```state->mav_mode = ...```, 
but rather encapsulate it in a function that implements the safety ```state_machine_try_arm()```